File: src/com/dafttech/terra/graphics/gui/Tooltip.java
package com.dafttech.terra.graphics.gui;

import com.dafttech.terra.game.world.Vector2;
import com.dafttech.terra.graphics.gui.anchors.AnchorMouse;
import com.dafttech.terra.graphics.gui.anchors.GUIAnchorSet;
import com.dafttech.terra.graphics.gui.elements.ElementLabel;
import com.dafttech.terra.graphics.gui.elements.GUIElement;

public class Tooltip {
    public static ElementLabel label;
    
    public static void init() {
        label = new ElementLabel(new Vector2(0, 0), "");
        
        GUIAnchorSet labelSet = new GUIAnchorSet();
        labelSet.addAnchor(new AnchorMouse());
        
        label.assignAnchorSet(labelSet);
    }
    
    public static GUIElement getLabel() {
        return label;
    }
    
    public static void setText(String txt) {
        label.setText(txt);
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/Tooltip.scala
package com.dafttech.terra.engine.gui

import com.dafttech.terra.engine.gui.anchors.{AnchorMouse, GUIAnchorSet}
import com.dafttech.terra.engine.gui.elements.{ElementLabel, GUIElement}
import com.dafttech.terra.engine.vector.Vector2d

object Tooltip {
  var label: ElementLabel = null

  def init: Unit = {
    label = new ElementLabel(Vector2d.Zero, "")
    val labelSet: GUIAnchorSet = new GUIAnchorSet
    labelSet.addAnchor(new AnchorMouse)
    label.assignAnchorSet(labelSet)
  }

  def getLabel: GUIElement = label

  def setText(txt: String): Unit = {
    label.setText(txt)
  }
}